### PR TITLE
tstime: add Append3339Nano

### PR DESCRIPTION
**Pull request by josharian**
And use it in log tail.

This eliminates the use of time.Format.
It shaves off ~9k from the iOS binary size.

Signed-off-by: Josh Bleecher Snyder <josh@tailscale.com>

### CODE REVIEW

File: tstime/tstime.go [hunk 1]
```diff
@@ -141,3 +142,71 @@ func Parse3339(s string) (time.Time, error) {
 func Parse3339B(b []byte) (time.Time, error) {
 	return parse3339m(mem.B(b))
 }
+
+func Append3339Nano(t time.Time, b []byte) []byte {
```
Comment (dsnet) [inline tstime/tstime.go:146]: Nit: perhaps AppendRFC3339Nano to make clean it's a specialized form of `time.RFC3339Nano`?

File: tstime/tstime.go [hunk 2]
```diff
@@ -141,3 +142,71 @@ func Parse3339(s string) (time.Time, error) {
...
+		b = append(b, '+')
+	}
+	zoneOffset /= 60
+	b = appendInt(b, int64(zoneOffset/60), 2)
+	b = append(b, ':')
+	b = appendInt(b, int64(zoneOffset%60), 2)
+	return b
+}
+
+func appendInt(b []byte, n int64, width int) []byte {
```
Comment (dsnet) [inline tstime/tstime.go:181]: Aside: I've seen wrapper logic to `strconv.AppendUint` to ensure a fixed width many times that i wonder if the strconv package itself should support this.

**Review by dsnet — APPROVED** [review]

File: tstime/tstime.go (hunk 2 above)
Comment (josharian) [reply tstime/tstime.go:181]: Yeah, that'd be nice.

**Comment by josharian** [comment]
Turns out that `-ldflags=-dumpdep` only shows you a single dependency. After cutting out this one...there are more. Plenty more. time.AppendFormat isn't going anywhere.

**Pull request closed without merging**